crates/modelardb_compression/src/types.rs

```diff
@@ -555,58 +557,263 @@ mod tests {
+            Value::MIN,
+            40.0,
+            5,
         );
+    }
 
-        assert_eq!(SWING_ID, model.model_type_id);
-        assert_eq!(uncompressed_timestamps.len() - 1, model.end_index);
-        assert_eq!(10.0, model.min_value);
-        assert_eq!(50.0, model.max_value);
-        assert_eq!(0, model.values.len());
+    #[test]
+    fn test_encoding_decoding_for_increasing_swing_with_residuals_with_new_maximum() {
+        let uncompressed_values = ValueArray::from(vec![10.0, 20.0, 30.0, 40.0, Value::MAX]);
+        test_encoding_decoding_for_swing(
+            uncompressed_values,
+            3,
+            10.0,
+            40.0,
+            0,
+            10.0,
+            Value::MAX,
+            5,
+        );
     }
 
     #[test]
-    fn test_model_attributes_for_decreasing_swing() {
-        let uncompressed_timestamps = TimestampArray::from(UNCOMPRESSED_TIMESTAMPS.to_vec());
+    fn test_encoding_decoding_for_increasing_swing_with_residuals_with_new_minimum_and_maximum() {
+        let uncompressed_values = ValueArray::from(vec![10.0, 20.0, 30.0, Value::MIN, Value::MAX]);
+        test_encoding_decoding_for_swing(
+            uncompressed_values,
+            2,
+            10.0,
+            30.0,
+            0,
+            Value::MIN,
+            Value::MAX,
+            8,
+        );
+    }
+
+    #[test]
+    fn test_encoding_decoding_for_decreasing_swing_without_residuals() {
         let uncompressed_values = ValueArray::from(vec![50.0, 40.0, 30.0, 20.0, 10.0]);
+        test_encoding_decoding_for_swing(uncompressed_values, 4, 10.0, 50.0, 1, 10.0, 50.0, 1);
+    }
+
+    #[test]
+    fn test_encoding_decoding_for_decreasing_swing_with_residuals_with_new_mininum() {
+        let uncompressed_values = ValueArray::from(vec![50.0, 40.0, 30.0, 20.0, Value::MIN]);
+        test_encoding_decoding_for_swing(
+            uncompressed_values,
+            3,
+            20.0,
+            50.0,
+            1,
+            Value::MIN,
+            50.0,
+            5,
+        );
+    }
+
+    #[test]
+    fn test_encoding_decoding_for_decreasing_swing_with_residuals_with_new_maximum() {
+        let uncompressed_values = ValueArray::from(vec![50.0, 40.0, 30.0, 20.0, Value::MAX]);
+        test_encoding_decoding_for_swing(
+            uncompressed_values,
+            3,
+            20.0,
+            50.0,
+            1,
+            20.0,
+            Value::MAX,
+            5,
+        );
+    }
+
+    #[test]
+    fn test_encoding_decoding_for_decreasing_swing_with_residuals_with_new_minimum_and_maximum() {
+        let uncompressed_values = ValueArray::from(vec![50.0, 40.0, 30.0, Value::MIN, Value::MAX]);
+        test_encoding_decoding_for_swing(
+            uncompressed_values,
+            2,
+            30.0,
+            50.0,
+            1,
+            Value::MIN,
+            Value::MAX,
+            8,
+        );
+    }
+
+    fn test_encoding_decoding_for_swing(
```

Comment:
I am not sure this is a convention we currently use consistently but maybe we should attempt to avoid naming test utility functions such as this with the "test_" prefix and only use this prefix for actual tests. Utility functions that use `assert!` could maybe be named `assert_` instead.

Reply:
Fixed.